### PR TITLE
Fix "failure rate" bugs

### PR DESCRIPTION
**Pull request by richsalz**
Fixes msan build break in Travis.
Fixes enable-crypto-mdebug build break.

### CODE REVIEW

**Comment by richsalz** [comment]
@kaduk you pinged me off-line, please take a look.

**Comment by levitte** [comment]
I restarted the Appveyor build, it seems it failed because github was inaccessible

**Comment by ekasper** [comment]
Can you explain what the bug was? Why would _msan_ care?

**Comment by richsalz** [comment]
The bug was that it was making malloc "fail" too often, and most places weren't checking for malloc returning NULL.

**Comment by kaduk** [comment]
@ekasper In particular the "roll > md_percent" vs "roll < md_percent" comparison.

**Comment by ekasper** [comment]
Where does this run on Travis and why would it affect `msan`? This stuff is behind `enable-crypto-mdebug`, isn't it?

File: crypto/mem.c [hunk 1]
```diff
@@ -104,7 +105,7 @@ static void parseit(void)
     /* Get the count (atol will stop at the @ if there), and percentage */
     md_count = atol(md_failstring);
     atsign = strchr(md_failstring, '@');
-    md_percent = atsign == NULL ? 100 : atoi(atsign + 1);
+    md_percent = atsign == NULL ? 0 : atoi(atsign + 1);
```
Comment (ekasper) [inline crypto/mem.c]: nit: it might help to call this `fail_percent` to avoid getting it backwards.

Comment (richsalz) [inline crypto/mem.c]: okay.  updated commit pushed.

**Comment by levitte** [comment]
> Where does this run on Travis and why would it affect `msan`? This stuff is behind `enable-crypto-mdebug`, isn't it?

I was wondering about that...  I think the msan stuff is something else...

**Comment by richsalz** [comment]
Ah, @ekasper you're right.  I forgot my build script has enable-crypto-mdebug always set....

**Comment by richsalz** [comment]
thanks!

**Pull request closed without merging**